### PR TITLE
Docs: the undo/redo guide.

### LINKED ISSUE #12301: Add Undo/Redo feature guide
Make a dedicated Undo/Redo feature guide

- [x] overview
- [x]  demo
- [x] keyboard shortcuts

### PR DESCRIPTION
**Pull request by godai78**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Docs: add the undo/redo guide. Closes #12301

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by Reinmar** [comment]
When documenting plugins make it clear that these are plugins. In the current proposal `UndoEditing` is described to be a class and `Undo` as a feature. `UndoUI` actually lacks any word :D

**Comment by Reinmar** [comment]
I left some inspiration in #12301.

**Review by apadol-cksource — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-undo/docs/features/undo-redo.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+category: features
+modified_at: 2022-08-29
+---
+
+# Undo/Redo support
```
[the remark this answers is not in the document]
Comment (godai78) [reply packages/ckeditor5-undo/docs/features/undo-redo.md]: Addressed in the commit.

**Review by apadol-cksource — APPROVED** [review]

File: packages/ckeditor5-undo/docs/features/undo-redo.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+The undo features lets the user withdraw recent changes done to the editor content, as well as bring them back again. All operations are remembered and organized info batches, that can later be easily undone or redone. By utilizing the batches approach, the feature is capable of selectively reverting past changes, not just the latest ones. This allows handling asynchronous actions such as image uploads without blocking the user from editing the document in the meantime.
```
[the remark this answers is not in the document]
Comment (godai78) [reply packages/ckeditor5-undo/docs/features/undo-redo.md]: Addressed in the commit.

**Pull request merged**